auth/service/api/v1/profile.go

```diff
@@ -0,0 +1,146 @@
+package v1
+
+import (
+	stdErrs "errors"
+	"net/http"
+
+	"github.com/ant0ine/go-json-rest/rest"
+
+	"github.com/tidepool-org/platform/request"
+	"github.com/tidepool-org/platform/service/api"
+	structValidator "github.com/tidepool-org/platform/structure/validator"
+	"github.com/tidepool-org/platform/user"
+)
+
+func (r *Router) ProfileRoutes() []*rest.Route {
+	return []*rest.Route{
+		rest.Get("/v1/users/:userId/profile", api.RequireAuth(r.GetProfile)),
+		rest.Get("/v1/users/legacy/:userId/profile", api.RequireAuth(r.GetLegacyProfile)),
+		// The following modification routes required custodian access in seagull, but I'm not sure that's quite right - it seems it should be if the user can modify the userId.
```

Comment:
This is very strange. I looked into seagull, and it seems that it really requires `custodian` permissions, but users can update their own profiles, so maybe gatekeeper returns `custodian` in the response when calling userInGroup with the same user for the grantor and grantee e.g. `/access/todd/todd`?

Reply:
The seagull code was very confusing with the layers of nesting but custodian check is *i believe*

* check if this is a service call, if yes allow.
* check if the requester in the request equals the user in the url param, if yes allow.
* check if the requester in the request has any permissions to the user in the url param OR the user in the url param has any permissions to the requester, if yes allow.
* otherwise disallow